### ISSUE TITLE
cts constructor comparing same tests fail

### ISSUE THREAD

**Issue by airlied** [triSYCL/triSYCL#161]
. line: 150
  . platform equality does not work correctly(comparing same)
  . line: 155
  . platform non-equality does not work correctly(comparing same)

I see this failure for a few of the object tests, which to me suggests it's a higher level problem and in how we construct the objects vs what the CTS tests/spec expect.

**Comment by keryell** [comment]
Can you send me some information about how you run the CTS, by the way?

**Comment by keryell** [comment]
Yes, very weird. After spending some time on it, it seems very strange,
Let's discuss this during the committee meeting https://gitlab.khronos.org/sycl/conformance-testsuite/issues/74

**Comment by keryell** [comment]
The issue was discussed during the committee but we need to dive further in the specification and discuss it again...
